src/libstd/rt/unwind/seh64_gnu.rs

```diff
@@ -0,0 +1,231 @@
+    BeginAddress: DWORD,
+    EndAddress: DWORD,
+    UnwindData: DWORD,
+}
+
+#[repr(C)]
+pub struct DISPATCHER_CONTEXT {
+    ControlPc: LPVOID,
+    ImageBase: LPVOID,
+    FunctionEntry: *const RUNTIME_FUNCTION,
+    EstablisherFrame: LPVOID,
+    TargetIp: LPVOID,
+    ContextRecord: *const CONTEXT,
+    LanguageHandler: LPVOID,
+    HandlerData: *const u8,
+    HistoryTable: *const UNWIND_HISTORY_TABLE,
+}
+
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub enum EXCEPTION_DISPOSITION {
+    ExceptionContinueExecution,
+    ExceptionContinueSearch,
+    ExceptionNestedException,
+    ExceptionCollidedUnwind
+}
+
+// From kernel32.dll
+extern "system" {
+    fn RaiseException(dwExceptionCode: DWORD,
+                      dwExceptionFlags: DWORD,
+                      nNumberOfArguments: DWORD,
+                      lpArguments: *const ULONG_PTR);
+
+    fn RtlUnwindEx(TargetFrame: LPVOID,
+                   TargetIp: LPVOID,
+                   ExceptionRecord: *const EXCEPTION_RECORD,
+                   ReturnValue: LPVOID,
+                   OriginalContext: *const CONTEXT,
+                   HistoryTable: *const UNWIND_HISTORY_TABLE);
+}
+
+#[repr(C)]
+struct PanicData {
+    data: Box<Any + Send + 'static>
+}
+
+pub unsafe fn panic(data: Box<Any + Send + 'static>) -> ! {
+    let panic_ctx = Box::new(PanicData { data: data });
+    let params = [Box::into_raw(panic_ctx) as ULONG_PTR];
+    rtdebug!("panic: ctx={:X}", params[0]);
+    RaiseException(RUST_PANIC,
+                   EXCEPTION_NONCONTINUABLE,
+                   params.len() as DWORD,
+                   &params as *const ULONG_PTR);
+    rtabort!("could not unwind stack");
+}
+
+pub unsafe fn cleanup(ptr: *mut c_void) -> Box<Any + Send + 'static> {
+    rtdebug!("cleanup: ctx={:X}", ptr as usize);
+    let panic_ctx = Box::from_raw(ptr as *mut PanicData);
+    return panic_ctx.data;
+}
+
+// SEH doesn't support resuming unwinds after calling a landing pad like libunwind does.
+// For this reason, MSVC compiler outlines landing pads into separate functions that
+// can be called directly from the personality function but are nevertheless able to find
+// and modify stack frame of the "parent" function.
+//
+// Since this cannot be done with libdwarf-style landing pads, rust_eh_personality instead
+// catches RUST_PANICs, runs the landing pad, then reraises the exception.
+//
+// Note that it makes certain assumptions about the exception:
+// 1. That RUST_PANIC is non-continuable, so no lower stack frame may choose to resume execution.
+// 2. That the first parameter of the exception is a pointer to an extra data area (PanicData).
+// Since these assumptions do not generally hold true for foreign exceptions (system faults,
+// C++ exceptions, etc), we make no attempt to invoke our landing pads (and, thus, destructors!)
+// for anything other than RUST_PANICs. This is considered acceptable, because the behavior of
+// throwing exceptions through a C ABI boundary is undefined.
```

Comment:
Could you format this to 80-chars wide? (what the rest of the runtime is formatted to)